version/main.go

```diff
@@ -31,8 +31,7 @@ func New(
 	inputs *dagger.Directory,
 	// +optional
 	// +defaultPath="/"
-	// +ignore=["*", "!.git", "!**/.gitignore", ".git/config"]
-	gitDir *dagger.Directory,
+	gitDir *dagger.GitRepository,
```

Comment:
Contextual git #10847 :tada:

This means we can checkout any ref, at any depth which is what we need for all the operations.